_universum/modules/launcher.py

```diff
@@ -85,6 +85,81 @@ def check_if_env_set(configuration):
     return True
 
 
+def check_str_match(string, expected_substrings, unexpected_substrings):
+    """The function to check whether specified string contains 'expected' and
+    does NOT contain 'unexpected' substrings.
+
+    >>> check_str_match("step 1", [], [])
+    True
+
+    >>> check_str_match("step 1", ["step 1"], [])
+    True
+
+    >>> check_str_match("step 1", ["step 1"], ["step 1"])
+    False
+
+    >>> check_str_match("step 1", [], ["step 1"])
+    False
+
+    >>> check_str_match("step 1", ["step "], ["1"])
+    False
+
+    :rtype: bool
+    """
+    result = False if expected_substrings else True
+    for substr in expected_substrings:
+        if substr in string:
+            result = True
+            break
+
+    for substr in unexpected_substrings:
+        if substr in string:
+            result = False
+            break
+    return result
+
+
+def get_match_patterns(filters):
+    """The function to parse 'filters' defined as a single string into a lists
```

Comment:
**`into the lists`**
please do not resolve conversations unless you fix them or comment on why the fix is not needed